### PR TITLE
feat: add websocket support for extended bidding on the lot page

### PR DESCRIPTION
**Pull request by mzikherman**
The type of this PR is: **Feature**

### Description

This uses websockets to listen for when a lot has had its time be extended. If so, that updated timestamp is used in place of the usual (`extendedBiddingEndAt ?? endAt`), and things feel like they're working pretty smoothly! I place a bid in one browser tab, and having the lot page open in a separate browser tab reveals that things update immedaitely.

Next step / improvement:
  - this doesn't properly share the websocket context (there are two actioncable subscriptions being used, whereas since they are for the same channel this should be memoized)
  - the code is sort of 🍝  among the two components, the websocket stuff can be made into a more proper context / hook

But, I think this is ok for now and mocktion purposes, the above is an enhancement. I was giving that a shot in https://github.com/artsy/force/commit/2c4710faa74dc7220fe9242e2ef6826cac3b7437 and pairing with @damassi .

### CODE REVIEW

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx [hunk 1]
```diff
@@ -30,7 +66,9 @@ export const LotTimer: React.FC<LotTimerProps> = ({ saleArtwork }) => {
     hasStarted,
     lotsAreClosing: false,
     isSaleInfo: false,
-    extendedBiddingEndAt,
+    extendedBiddingEndAt: isExtended
```
Comment (bhoggard) [inline src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx:69]: What is the purpose of the ternary operator here?

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx [hunk 2]
```diff
@@ -5,22 +5,58 @@ import { Text, Spacer, Box } from "@artsy/palette"
...
+
+  useEffect(() => {
+    if (extendedBiddingIntervalMinutes) {
+      const actionCable = require("actioncable")
+      const CableApp = {} as any
+      CableApp.cable = actionCable.createConsumer(
+        getENV("GRAVITY_WEBSOCKET_URL")
+      )
+
+      CableApp.cable.subscriptions.create(
```
Comment (nspinazz89) [inline src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx:39]: It's incredibly cool that this is all it takes to setup websockets. I would have thought there was going to be way more overhead!

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx (hunk 1 above)
Comment (nspinazz89) [reply src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx:69]: yeah a little confused by this as well

**Review by nspinazz89 — APPROVED** [review]
Looks good to me!

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx (hunk 1 above)
Comment (mzikherman) [reply src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx:69]: `extendedBiddingEndAt` can be in the past when the lot is extended the first time (after you load the page), so the timer label doesn't shift to show `Extended: ...` when that occurs. Reloading the page is fine, the helper needs to know the websocket-updated time for that scenario.

Comment (nspinazz89) [reply src/v2/Apps/Artwork/Components/ArtworkSidebar/LotTimer.tsx:69]: ohhhh I see that makes sense!

**Comment by mzikherman** [comment]
Gonna merge so we can try it out , but def should follow up on having it use React context properly for this.

**Pull request merged**